### PR TITLE
Sebastian subdivide

### PR DESCRIPTION
**Pull request by sebwolf-de**
This script enables us to subdivide a (volume) mesh, by replacing each tetrahedron with 8 sub tetrahedrons.
Note that I use a slightly different seissolxdmf writer than @Thomas-Ulrich created in [SeisSol/Meshing #6](https://github.com/SeisSol/Visualization/pull/6).
@krenzland and I tested this script for our Helsinki scenarios.

### CODE REVIEW

**Comment by Thomas-Ulrich** [comment]
Hi, 
I've tested with the latest version of the seissolxdmfwriter,
Input file
```xml
<?xml version="1.0" ?>
<!DOCTYPE Xdmf SYSTEM "Xdmf.dtd" []>
<Xdmf Version="2.0">
 <Domain>
  <Grid Name="puml mesh" GridType="Uniform">
   <Topology TopologyType="Tetrahedron" NumberOfElements="2404028">
    <DataItem NumberType="Int" Precision="8" Format="HDF" Dimensions="2404028 4">marmara_mesh.puml.h5:/connect</DataItem>
   </Topology>
   <Geometry name="geo" GeometryType="XYZ" NumberOfElements="462242">
    <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="462242 3">marmara_mesh.puml.h5:/geometry</DataItem>
   </Geometry>
   <Attribute Name="group" Center="Cell">
    <DataItem  NumberType="Int" Precision="4" Format="HDF" Dimensions="2404028">marmara_mesh.puml.h5:/group</DataItem>
   </Attribute>
   <Attribute Name="boundary" Center="Cell">
    <DataItem NumberType="Int" Precision="4" Format="HDF" Dimensions="2404028">marmara_mesh.puml.h5:/boundary</DataItem>
   </Attribute>
  </Grid>
 </Domain>
</Xdmf>
```
output file
```xml
<?xml version="1.0" ?>
<!DOCTYPE Xdmf SYSTEM "Xdmf.dtd" []>
<Xdmf Version="2.0">
 <Domain>
  <Grid Name="TimeSeries" GridType="Collection" CollectionType="Temporal">
   <Grid Name="step_0" GridType="Uniform">
    <Topology TopologyType="Tetrahedron" NumberOfElements="19232224">
     <DataItem NumberType="Int" Precision="8" Format="HDF" Dimensions="19232224 4">/home/ulrich/trash/marmara_mesh-subdivided.h5:/connect</DataItem>
    </Topology>
    <Geometry name="geo" GeometryType="XYZ" NumberOfElements="3354608">
     <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="3354608 3">/home/ulrich/trash/marmara_mesh-subdivided.h5:/geometry</DataItem>
    </Geometry>
    <Time Value="0.0"/>
    <Attribute Name="boundary" Center="Cell">
     <DataItem ItemType="HyperSlab" Dimensions="19232224">
      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">0 0 1 1 1 19232224</DataItem>
      <DataItem NumberType="UInt" Precision="4" Format="HDF" Dimensions="1 19232224">/home/ulrich/trash/marmara_mesh-subdivided.h5:/boundary</DataItem>
     </DataItem>
    </Attribute>
    <Attribute Name="group" Center="Cell">
     <DataItem ItemType="HyperSlab" Dimensions="19232224">
      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">0 0 1 1 1 19232224</DataItem>
      <DataItem NumberType="UInt" Precision="4" Format="HDF" Dimensions="1 19232224">/home/ulrich/trash/marmara_mesh-subdivided.h5:/group</DataItem>
     </DataItem>
    </Attribute>
   </Grid>
  </Grid>
 </Domain>
</Xdmf>
```
note the inconsistent xml file format.
So you may need to copy and modify the seissolxdmfwriter to get a suitable format for seissol.

**Comment by krenzland** [comment]
Hm, that is in principle not a problem. The file can be opened in paraview (right, @sebwolf-de) and SeisSol ignores the xdmf file anyway. Would be nicer with the proper xdmf file though.

**Comment by sebwolf-de** [comment]
First review https://github.com/SeisSol/Visualization/pull/7

File: subdivide_mesh/subdivide_mesh.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Read xdmf mesh and subdivide each element into eight subelements"
+    )
+    parser.add_argument("filename", help="xdmf mesh file")
+    args = parser.parse_args()
+
+    parse_filename = re.compile("(.+)\.xdmf")
+
+    filename_prefix = parse_filename.match(args.filename)
```
Comment (Thomas-Ulrich) [inline subdivide_mesh/subdivide_mesh.py:149]: would be better to use os.basename and os.path.splitext IMO.
currently, If I run:
```
./subdivide_mesh.py ~/trash/marmara_mesh.xdmf
``` 
Then the full path is written in the xdmf.

**Review by Thomas-Ulrich — COMMENTED** [review]
Very nice and readable code. Just a small comment.

**Review by Thomas-Ulrich — APPROVED** [review]

**Pull request merged**